Fix part of #3690 : Wiki update - How to run static CI check locally

MohitGupta121 opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation

Fix part of #3690 : Wiki Update - How to run static CI check locally

<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

MohitGupta121 committed:
static ci checks run locally

MohitGupta121 reviewed (commented):
@adhiamboperes PTAL, Thanks. Please suggest if something need to change.

MohitGupta121 committed:
Merge branch 'develop' into static_check_locally_wiki

MohitGupta121 committed:
fix todo tests fail

MohitGupta121 committed:
Merge remote-tracking branch 'upstream/static_check_locally_wiki' into static_check_locally_wiki

adhiamboperes commented:
Hi @MohitGupta121, why are there changes in the todo line exemptions file? Did you add/remove a todo?

MohitGupta121 commented:
> Hi @MohitGupta121, why are there changes in the todo line exemptions file? Did you add/remove a todo?

No @adhiamboperes basically these wiki Todos are exception. So I added new line in TOC due to which these todos lines get shifted to next lines and the CI checks fail that's why I added new line number for correct TODO.

adhiamboperes commented:
> > Hi @MohitGupta121, why are there changes in the todo line exemptions file? Did you add/remove a todo?
> 
> No @adhiamboperes basically these wiki Todos are exception. So I added new line in TOC due to which these todos lines get shifted to next lines and the CI checks fail that's why I added new line number for correct TODO.

Got it, thanks. This LGTM

seanlip commented on wiki/Static-Analysis-Checks.md:
```diff
@@ -227,3 +228,11 @@ To fix the failures for this check: resolve the TODO items and then close the is
+
+# How to run static checks locally
+
+- Go to failing CI check in your GitHub PR.
```
failing --> the failing

MohitGupta121 replied:
Done.

seanlip commented on wiki/Static-Analysis-Checks.md:
```diff
@@ -227,3 +228,11 @@ To fix the failures for this check: resolve the TODO items and then close the is
+# How to run static checks locally
+
+- Go to failing CI check in your GitHub PR.
+- Scroll failing CI check logs to the top and find the Bazel command that was run for this script.
```
Scroll to the top of the failing CI check logs, and find the ...

MohitGupta121 replied:
Done.

seanlip commented on wiki/Static-Analysis-Checks.md:
```diff
@@ -227,3 +228,11 @@ To fix the failures for this check: resolve the TODO items and then close the is
+
+- Go to failing CI check in your GitHub PR.
+- Scroll failing CI check logs to the top and find the Bazel command that was run for this script.
+- Alternatively, in android studio, go to the `.github` folder and find the [static_checks.yml](https://github.com/oppia/oppia-android/blob/develop/.github/workflows/static_checks.yml) file. Search for the line that corresponds to the name of the job that failed, and you can run the same script on your local terminal.
```
android studio --> Android Studio

...job that failed. You can then run the ...

MohitGupta121 replied:
Done.

seanlip commented on wiki/Static-Analysis-Checks.md:
```diff
@@ -227,3 +228,11 @@ To fix the failures for this check: resolve the TODO items and then close the is
+- Scroll failing CI check logs to the top and find the Bazel command that was run for this script.
+- Alternatively, in android studio, go to the `.github` folder and find the [static_checks.yml](https://github.com/oppia/oppia-android/blob/develop/.github/workflows/static_checks.yml) file. Search for the line that corresponds to the name of the job that failed, and you can run the same script on your local terminal.
+
+Note: Before running script command in your local terminal make sure you have Bazel installed. To know how to setup Bazel for oppia-android follow these [instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions).
```
Note: Before running the script command in your local terminal, make sure you have Bazel installed. To learn how to set up Bazel for Oppia Android, follow these ...

MohitGupta121 replied:
Done.

seanlip reviewed (changes requested):
@MohitGupta121 Suggested some small grammar changes, PTAL. Thanks!

MohitGupta121 committed:
review changes updated

MohitGupta121 committed:
Merge branch 'develop' into static_check_locally_wiki

MohitGupta121 commented:
@seanlip PTAL, make the required changes. Thanks!

seanlip reviewed (approved):
@MohitGupta121 LGTM. Thanks!
